pmlc/dialect/stripe/rewrites.cc

```diff
@@ -0,0 +1,186 @@
+// Copyright 2019, Intel Corporation
+
+#include "pmlc/dialect/stripe/rewrites.h"
+
+#include "mlir/IR/PatternMatch.h"
+#include "pmlc/dialect/stripe/analysis.h"
+#include "pmlc/dialect/stripe/dialect.h"
+#include "pmlc/dialect/stripe/ops.h"
+#include "pmlc/dialect/stripe/transforms.h"
+
+namespace pmlc::dialect::stripe {
+
+mlir::PatternMatchResult SimplifyPoly::matchAndRewrite(AffinePolyOp op, mlir::PatternRewriter& rewriter) const {
+  bool has_non_canonical = false;
+  for (size_t i = 0; i < op.getNumOperands(); i++) {
+    if (op.getOperand(i)->getKind() != Value::Kind::BlockArgument) {
+      has_non_canonical = true;
+    }
+  }
+  AffinePolynomial a(op.result());
+  if (a.constant == 0 && a.terms.size() == 1 && a.terms.begin()->second == 1) {
+    rewriter.replaceOp(op, a.terms.begin()->first);
+    return matchSuccess();
+  }
+  if (has_non_canonical) {
+    rewriter.replaceOpWithNewOp<AffinePolyOp>(op, a);
+    return matchSuccess();
+  }
+  return matchFailure();
+}
+
+mlir::PatternMatchResult SimplifyNopRefines::match(RefineOp op) const {
+  for (auto* offset : op.offsets()) {
+    if (AffinePolynomial(offset) != AffinePolynomial()) {
+      return matchFailure();
+    }
+  }
+  if (op.getAttr(Dialect::getStripeAttrsName())) {
+    return matchFailure();
+  }
+  return matchSuccess();
+}
+
+void SimplifyNopRefines::rewrite(RefineOp op, mlir::PatternRewriter& rewriter) const {  // NOLINT(runtime/references)
+  rewriter.replaceOp(op, op.in());
+}
+
+void InlineNoIndexParallelFors::rewrite(ParallelForOp op, mlir::PatternRewriter& rewriter) const {
+  auto oblock = op.getOperation()->getBlock();
+  auto iblock = &op.inner().front();
+  oblock->getOperations().splice(Block::iterator(op), iblock->getOperations(), iblock->begin(),
+                                 std::prev(iblock->end(), 1));
+  rewriter.eraseOp(op);
```

Comment:
Should we use `rewriter.replaceOp(op, llvm::None)`?

Reply:
Nope, eraseOp is newly added and more correct I believe.